### PR TITLE
Added software license item recipients

### PR DESCRIPTION
**Pull request by zombigiraffe**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #13774

### CODE REVIEW

**Review by cconard96 — APPROVED** [review]

**Review by cedric-anne — CHANGES_REQUESTED** [review]
To be able to send notifications to technician/group in charge, we have to send separate notifications for each licence. Indeed, the default logic expects `target_object` property to be unique `CommonDBTM` object (see `NotificationTarget::addUserByField()`).

Splitting notification has been done recently for `Domain` and `Certificate` classe to be able to handle these targets, see #11068.

File: src/NotificationTargetSoftwareLicense.php [hunk 1]
```diff
@@ -40,21 +40,51 @@ class NotificationTargetSoftwareLicense extends NotificationTarget
...
+            if (!$license->isNewItem()) {
+                $options['licenses'][] = $license->fields;// Compatibility with old behaviour
+            }
+        } else {
+            Toolbox::deprecated('Using "licenses" option in NotificationTargetSoftwareLicense is deprecated.');
+        }
+        if (!isset($options['entities_id'])) {
+            $options['entities_id'] = $license->fields['entities_id'];
+        } else {
+            Toolbox::deprecated('Using "entities_id" option in NotificationSoftwareLicense is deprecated.');
```
Comment (cconard96) [inline src/NotificationTargetSoftwareLicense.php:64]:
```suggestion
            Toolbox::deprecated('Using "entities_id" option in NotificationTargetSoftwareLicense is deprecated.');
```

**Review by cconard96 — CHANGES_REQUESTED** [review]

**Comment by cedric-anne** [comment]
I changed target branch from 10.0/bugfixes to main.

**Comment by cedric-anne** [comment]
@orthagh 

We talked yesterday about the same change for `Cartridge`/`Consumable` alerts. You said that having distinct notifications could result in sending way too much notifications. Do you consider the issue will be the same for expired `SoftwareLicence` alerts?

**Comment by orthagh** [comment]
I don't know particulary for this itemtype, but this fact is we should change (at least not in bugfixes) how much notifications we push for one event.
But as this seems to be asked frequently, we may change it main and if possible maybe permits to choose the behavior (grouped or one send per item)

**Comment by cedric-anne** [comment]
> if possible maybe permits to choose the behavior (grouped or one send per item)

If we group items, we have to keep in mind that we have to prevent sending to someone a notification about an item he is not supposed to be notified. So we have to only group items from same entity, with same tech in charge, same froup in charge, and probably more.

**Comment by cedric-anne** [comment]
I am closing this topic because it is a too complex topic that requires additional functional and technical specifications that we do not have time to do at the moment.

**Pull request closed without merging**